pkgs/c_compiler/test/cbuilder/cbuilder_cross_android_test.dart

```diff
@@ -91,4 +77,64 @@ void main() {
       });
     }
   }
+
+  test('Cbuilder API levels binary difference', () async {
+    const target = Target.androidArm64;
+    const linkMode = LinkMode.dynamic;
+    const apiLevel1 = flutterAndroidNdkVersionLowestSupported;
+    const apiLevel2 = flutterAndroidNdkVersionHighestSupported;
+    await inTempDir((tempUri) async {
+      final out1Uri = tempUri.resolve('out1/');
+      final out2Uri = tempUri.resolve('out2/');
+      final out3Uri = tempUri.resolve('out3/');
+      await Directory.fromUri(out1Uri).create();
+      await Directory.fromUri(out2Uri).create();
+      await Directory.fromUri(out3Uri).create();
+      final lib1Uri = await buildLib(out1Uri, target, apiLevel1, linkMode);
+      final lib2Uri = await buildLib(out2Uri, target, apiLevel2, linkMode);
+      final lib3Uri = await buildLib(out3Uri, target, apiLevel2, linkMode);
+      final bytes1 = await File.fromUri(lib1Uri).readAsBytes();
+      final bytes2 = await File.fromUri(lib2Uri).readAsBytes();
+      final bytes3 = await File.fromUri(lib3Uri).readAsBytes();
+      // Different API levels should lead to a different binary.
+      expect(bytes1.equals(bytes2), false);
+      // Identical API levels should lead to an identical binary.
+      expect(bytes2.equals(bytes3), true);
```

Comment:
nit: I think `expect` automatically checks for equality of `Iterables`: https://github.com/dart-lang/matcher/blob/7e1011772566a8d1817c725a71f091e2791721c8/lib/src/equals_matcher.dart#L15

So it can be 
```
expect(bytes1, isNot(bytes2));
expect(bytes2, bytes3);
```

Reply:
Thanks @HosseinYousefi!

I tried `notEquals` instead of `isNot`, but that got me something in `dart2js`. 🙈